lib/msal-browser/test/interaction_client/BaseInteractionClient.spec.ts

```diff
@@ -117,5 +117,77 @@ describe("BaseInteractionClient", () => {
             expect(pca.getAccountByHomeId(testAccountInfo2.homeAccountId)).toMatchObject(testAccountInfo2);
             expect(pca.getActiveAccount()).toBe(null);
         });
+
+    });
+    describe("validateRequestAuthority()", () => {
+        let testAccountInfo1: AccountInfo;
+
+        beforeEach(() => {
+            const testIdTokenClaims: TokenClaims = {
+                "ver": "2.0",
+                "iss": "https://login.microsoftonline.com/9188040d-6c67-4c5b-b112-36a304b66dad/v2.0",
+                "sub": "AAAAAAAAAAAAAAAAAAAAAIkzqFVrSaSaFHy782bbtaQ",
+                "name": "Abe Lincoln",
+                "preferred_username": "AbeLi@microsoft.com",
+                "oid": "00000000-0000-0000-66f3-3332eca7ea81",
+                "tid": "3338040d-6c67-4c5b-b112-36a304b66dad",
+                "nonce": "123523",
+            };
+
+            testAccountInfo1 = {
+                homeAccountId: TEST_DATA_CLIENT_INFO.TEST_HOME_ACCOUNT_ID,
+                localAccountId: TEST_DATA_CLIENT_INFO.TEST_UID,
+                environment: "login.windows.net",
+                tenantId: testIdTokenClaims.tid || "",
+                username: testIdTokenClaims.preferred_username || ""
+            };
+
+            const testAccount1: AccountEntity = new AccountEntity();
+            testAccount1.homeAccountId = testAccountInfo1.homeAccountId;
+            testAccount1.localAccountId = testAccountInfo1.localAccountId;
+            testAccount1.environment = testAccountInfo1.environment;
+            testAccount1.realm = testAccountInfo1.tenantId;
+            testAccount1.username = testAccountInfo1.username;
+            testAccount1.name = testAccountInfo1.name;
+            testAccount1.authorityType = "MSSTS";
+            testAccount1.clientInfo = TEST_DATA_CLIENT_INFO.TEST_CLIENT_INFO_B64ENCODED;
+
+            pca.setActiveAccount(testAccountInfo1);
+            // @ts-ignore
+            pca.browserStorage.setAccount(testAccount1);
+        });
+
+        afterEach(() => {
+            window.sessionStorage.clear();
+        });
+
+        it("Throw error when authority in request or MSAL config does not match with environment set for account", async () => {
+            let loginRequest = {
+                authority: "https://login.windows-ppe.net/common",
+                account: pca.getActiveAccount()
+            };
+
+            if(loginRequest.account) {
+                await testClient.validateRequestAuthority(loginRequest.authority, loginRequest.account)
+                    .catch(error => {
```

Comment:
If this method doesn't throw this test will still pass. Either wrap this in a try/catch instead of using promise chaining. Or use the done() method instead of async/await